include/osmium/thread/pool.hpp

```diff
@@ -162,7 +162,11 @@ namespace osmium {
 
             static constexpr int default_num_threads = 0;
 
-            static Pool& instance() {
+            static std::unique_ptr<Pool> create_instance(int num_threads = default_num_threads) {
+                return std::unique_ptr<Pool>(new Pool(num_threads, detail::get_work_queue_size()));
+            }
```

Comment:
> Do we need this function?

No, unless `Pool` ctor must be non-public, what I assumed.

I also aimed for some interface symmetry, more explicit and self-describing that one is supposed to use either `default_instance`  or `create_instance`, not mixing them.

If you prefer, I can remove it and make `Pool` ctor public. Let me know.